.buildkite/scripts/check_do_docker.sh

```diff
@@ -0,0 +1,15 @@
+#!/bin/bash
+
+set -exuo pipefail
+
+# This script checks if we should do a docker build, and if so adds a step in
```

Comment:
nit: mind adding some kind of check to ensure this is only attempted in on a CI host - something like `[[ -z BUILDKITE ]] || (echo "..."; exit 1)`